crates/autopilot/src/infra/solvers/dto/solve.rs

```diff
@@ -47,6 +48,13 @@ impl Request {
                 .unique_by(|token| token.address)
                 .collect(),
             deadline: Utc::now() + chrono::Duration::from_std(time_limit).unwrap(),
+            surplus_capturing_jit_order_owners: (surplus_capturing_jit_order_owners.is_empty())
```

Comment:
This should encode values if there are values to encode.
```suggestion
            surplus_capturing_jit_order_owners: (!surplus_capturing_jit_order_owners.is_empty())
```
But overall I think it's probably not worth the effort to not serialize empty vectors anyway. So the `Option<Vec<T>>` in the request might as well be `Vec<T>`.
And if we were keen on not serializing empty vectors we could do:
```rust
#[serde(skip_serializing_if = "Vec::is_empty")]
```
This still makes sense because vectors in rust actually only allocate the first time you push into them making the creation of a bunch of empty vectors basically free.
But yeah overall introducing `Option` here and in the deserializing logic just results in more conversion logic for not much benefit.